src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -228,41 +228,46 @@ class IndexManagerTests extends SparkFunSuite with SparkInvolvedSuite {
       schema: StructType,
       df: DataFrame,
       state: String = Constants.States.ACTIVE): IndexLogEntry = {
-    val serializedPlan = LogicalPlanSerDeUtils.serialize(df.queryExecution.logical, spark)
-    val sourcePlanProperties = SparkPlan.Properties(
-      serializedPlan,
-      LogicalPlanFingerprint(
-        LogicalPlanFingerprint.Properties(
-          Seq(Signature(
-            LogicalPlanSignatureProvider.create().name,
-            LogicalPlanSignatureProvider.create().signature(df.queryExecution.optimizedPlan))))))
-    val sourceFiles = df.queryExecution.optimizedPlan.collect {
-      case LogicalRelation(
+
+    LogicalPlanSignatureProvider.create().signature(df.queryExecution.optimizedPlan) match {
+      case Some(s) =>
+        val serializedPlan = LogicalPlanSerDeUtils.serialize(df.queryExecution.logical, spark)
+        val sourcePlanProperties = SparkPlan.Properties(
+          serializedPlan,
+          LogicalPlanFingerprint(
+            LogicalPlanFingerprint.Properties(
+              Seq(Signature(
+                LogicalPlanSignatureProvider.create().name,
+                s)))))
+        val sourceFiles = df.queryExecution.optimizedPlan.collect {
+          case LogicalRelation(
           HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
```

Comment:
indentation looks off? is this from auto-format?

Reply:
fixed.